lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -205,8 +205,8 @@ constFoldFIRRTLBinaryOp(Operation *op, ArrayRef<Attribute> operands,
   if (resultType.getWidthOrSentinel() < 0)
     return {};
 
-  // Any binary non-cmp op on I<0> is 0.
-  if (resultType.getWidthOrSentinel() == 0 && opKind == BinOpKind::Compare)
```

Comment:
opKind != BinOpKind::Compare

Comparisons should return an i1 with cmp(0,0)

Reply:
Comparison return type is always i1, so the check seemed redundant but happy to add it!

Reply:
Is the entire check unnecessary?  Skimming though the function, APInt 0-width support might make it redundant.

Reply:
Nope, this is needed to canonicalize the operations in the added test (result is zero-width but not both operands are zero-width).